### ISSUE TITLE
Regarding SCAN loss

### ISSUE THREAD

**Issue by Stupack** [wvangansbeke/Unsupervised-Classification#6]
Hi, 

I was checking your code for the SCAN loss and it states:

    def forward(self, anchors, neighbors):
        """
        input:
            - anchors: logits for anchor images w/ shape [b, num_classes]
            - neighbors: logits for neighbor images w/ shape [b, num_classes]
        output:
            - Loss
        """

Im not sure If Im missing the obvious, but I don't understand why the neighbors is of shape [batch, n_classes]. If we consider a batch of images of [128, n_classes] with 5 neighbors for each data point, shouldn't the batch be [k*batch, n_classes]? Is this a memory strat to deal with large k numbers?

Thanks.

**Comment by wvangansbeke** [comment]
Hi @Stupack,

Actually a very similar question as in #4. "We sample uniformly from the K nearest neighbors during training. Therefore, it is highly likely that the anchor sees a different neighbor in the next epoch. So, if you train long enough it should have the exact same effect as Eq. 2 (in the paper). After all, it is not practical to include a lot of neighbors for every sample during a forward pass, since this does not scale well with the amount K." 

So indeed, we only sample 1 of the nearest neighbors to prevent potential scaling or balancing issues. That's why the input size is [batch_size, num_classes].

Hope this helps,
Wouter
